pkg/alertmanager/state_persister.go

```diff
@@ -94,6 +110,7 @@ func (s *statePersister) persist(ctx context.Context) error {
 
 	fs, err := s.state.GetFullState()
 	if err != nil {
+		s.persistFailed.Inc()
```

Comment:
The current logic is fine, but typically it's more robust if you give a name to the returned `error` and then define a `defer` which increase the failed metric if the returned `error != nil`